src/js/src/mixins/HistoryRoutingUtils.js

```diff
@@ -2,10 +2,34 @@ export default {
   methods: {
     $_pushSearchHistory(destination, query, appliedFacets, solrSettings) {
     // Should be deleted before production.
-    console.log('We pushed a history.')
-    let newFacetUrl = appliedFacets !== '' ? '&facets=' + encodeURIComponent(appliedFacets) : ''
-    let newOptionsUrl = '&offset=' + solrSettings.offset + '&grouping=' + solrSettings.grouping + '&imgSearch=' + solrSettings.imgSearch + '&urlSearch=' + solrSettings.urlSearch
-    history.pushState({name: destination}, destination, '?q=' + query + newFacetUrl + newOptionsUrl)
+    //console.log('We about to push a history.', appliedFacets)
+    let currentQuery = this.$router.history.current.query
+    if(query !== '' && query !== undefined && (currentQuery.query !== query ||
+       currentQuery.grouping !== solrSettings.grouping ||
+       currentQuery.urlSearch !== solrSettings.urlSearch ||
+       currentQuery.imgSearch !== solrSettings.imgSearch ||
+       currentQuery.fileSearch !== solrSettings.fileSearch ||
+       currentQuery.facets !== appliedFacets.join('') ||
+       currentQuery.offset !== solrSettings.offset)) {
+      this.$router.push({name: destination, query: { query: query,
+                                                     grouping: solrSettings.grouping,
+                                                     imgSearch: solrSettings.imgSearch,
+                                                     offset: solrSettings.imgSearch === true ? 0 : solrSettings.offset,
+                                                     urlSearch: solrSettings.urlSearch,
+                                                     fileSearch: solrSettings.fileSearch,
+                                                     facets:appliedFacets.join('')
+                                                     } })
+      //console.log(this.$router, 'we pushed route yey')
+      }
+      else {
+        //console.log('No changes, no dice!')
+      }
```

Comment:
What is this?

Reply:
Teehee, leftover for when I was testing. Sorry :D